README.md

```diff
@@ -356,12 +338,20 @@ MemoryLocation can be used if you don't want router to touch the address bar at
 e.g.
 
 ```js
-var cherrytree = require('cherrytree')
-var MemoryLocation = require('cherrytree/lib/locations/memory')
-var router = cherrytree()
-routerlisten(new MemoryLocation())
+var router = cherrytree({
+  location: 'memory'
```

Comment:
Yeah, that name is very old, I don't even know exactly why it's called `location`..

It comes from here, originally:
https://github.com/emberjs/ember.js/blob/bc84326936de4d7d3305a33017c84900e7905331/packages/ember-routing/lib/main.js#L12-L16

Looking at https://github.com/rackt/react-router/blob/c43fb617fbdcf0fcaa5f0aeb4488cec1e7cec779/docs/guides/basics/Histories.md, they seem to be calling the thing that provides you with the location `history` and the `location` is then an object describing the location.

Ah! Location probably comes from `document.location` :) Ok, I see your point..

So I'm thinking either `locationProvider` or `history`. I could do the following renames:

* `location: 'memory'` -> `history: 'memory'`
* `location: 'history'` -> `history: 'browser'`
* `lib/locations/history.js` -> `lib/history/browser.js`
* `lib/locations/memory.js` -> `lib/history/memory.js`

On the other hand.. the way `document.location` can tell you the URL via `href`, so can `lib/location/memory.js` via `getURL` function. So you could see them as locations. In fact, in this case, I would only do a single rename:

* `lib/locations/history` -> `lib/locations/browser`